src/CodeFixes/CSharp/CodeFixes/DeclareAsNullableCodeFixProvider.cs

```diff
@@ -85,4 +90,42 @@ private static void TryRegisterCodeFix(CodeFixContext context, Diagnostic diagno
 
         context.RegisterCodeFix(codeAction, diagnostic);
     }
+
+    private static void TryRegisterCodeFixForCast(CodeFixContext context, Diagnostic diagnostic, TypeSyntax type)
+    {
+        if (type.IsKind(SyntaxKind.NullableType))
+            return;
+
+        CodeAction codeAction = CodeAction.Create(
+            "Declare as nullable",
+            async ct =>
+            {
+                NullableTypeSyntax newType = SyntaxFactory.NullableType(type.WithoutTrivia()).WithTriviaFrom(type);
+                Document changedDocument = await context.Document.ReplaceNodeAsync(type, newType, ct).ConfigureAwait(false);
+
+                // This could be in a variable declaration, so grab the new syntax root and find the newly-replaced type node
+                SyntaxNode root = await changedDocument.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+                SyntaxNode insertedNewType = root.FindNode(type.Span);
+
+                // Try finding a surrounding variable declaration whose type we also have to change
+                if (insertedNewType.AncestorsAndSelf().FirstOrDefault(a => a.IsKind(SyntaxKind.EqualsValueClause)) is EqualsValueClauseSyntax equalsValueClause)
```

Comment:
I'm not sure about using `AncestorsAndSelf`. I would just check if parent of cast expression is assigned value (`Initializer.Value`).

Reply:
The thought was to automatically handle perhaps a ParenthesizedExpression around, but yeah, it's probably rare enough to have those in spades around initializers. Especially with the added risk of ancestor-ing through unexpected intermediary expressions. Will change.